src/logger.rs

```diff
@@ -6,7 +6,7 @@ use librocksdb_sys::{DBEnv, DBInfoLogLevel as InfoLogLevel, DBLogger};
 use std::ffi::{CString, VaList};
 
 pub trait Logger: Send + Sync {
-    fn logv(&self, log_level: InfoLogLevel, format: *const c_char, ap: VaList);
+    unsafe fn logv(&self, log_level: InfoLogLevel, format: *const c_char, ap: VaList);
```

Comment:
does changing `format` from `*const c_char` to `std::ffi::CStr` help get rid of the use of `unsafe`?